guppylang/compiler/core.py

```diff
@@ -27,6 +49,7 @@ class CompiledGlobals:
     checked: dict[DefId, CheckedDef]
     compiled: dict[DefId, CompiledDef]
     worklist: set[DefId]
+    compiler_functions: dict[str, CustomCompilerFunction]
```

Comment:
I'd prefer to just store a dict of `hugr.build.function.Function`s and leave the call logic to whoever wants to use this function.

Maybe even just store `Wire`s corresponding to the node, then we could use the same system to store global constants and stuff like that?